rfcs/text/0000-roadmaps+planning.md

```diff
@@ -0,0 +1,421 @@
+
+#### Modeling progress with ZenHub Pipelines [*dcp* repo]
+
+New items in the *dcp* repo appear in the ZenHub *New* pipeline and MUST be triaged into the *Product Backlog* pipeline, the *Icebox* pipeline, or the *Closed* pipeline by the Release Shepherd or Product Owners.
+
+The ZenHub *Icebox* pipeline SHOULD be used with great restraint in the *dcp* repo. Its explicit purpose is to queue up potential futures from the DCP roadmap such as pending DevSecOps issues. Individual component repo(s)  MAY use the *Icebox* pipeline in a manner of their own choosing.
+
+No issues in the *dcp* repo MUST be assigned to the *Epic* or *Sprint Backlog* pipelines. Individual component repo(s) MAY use these pipelines in a manner of their own choosing.
+
+At the level of the *dcp* repo, a subset of ZenHub pipelines are used to track progress. All Epics or issues in a Release are initially assigned to the *Product Backlog* pipeline. Once the Epic (or any of its children) is started, then its *dcp* issue is moved to the *In Progress* pipeline. When all the children of the Epic are in the *Closed* pipeline, then the *dcp* issue is moved to the *Closed* pipeline.
+
+#### Modeling progress with ZenHub Pipelines [component repo]
+
+When an issue is started, it is moved to the *In Progress* pipeline.
+
+When a pull request is filed and in review for an *In Progress* item, the issue is moved to the *Review* pipeline.
+
+If the pull request is rejected, the issue returns to *In Progress*; otherwise, it is merged and the related issue moves to the *Merged* Pipeline.
+
+When all changes are verified in [Production](https://allspark.dev.data.humancellatlas.org/dcp-ops/docs/wikis/SOP:%20Releasing%20new%20Versions%20of%20DCP%20Software), the issue is moved to the *Closed* pipeline.
+
+### Release Demonstration and Retrospective
+
+**At Release Completion**. The Release Shepherd MUST schedule and faciliate a Release Demonstration and Retrospective meeting with the DCP Community and Oversight. It is RECOMMENDED that this occur during a DCP F2F when schedules align. 
+
+The intent of the meeting is to demonstrate the Release deliverables as documented in the Roadmap Objectives, to share learnings, and identify opportunities to improve the planning process for future Releases. 
+
+### Acceptance Criteria
+
+This process will be under continuous community introspection and improvement through Retrospectives. If the process fails to deliver product increments with high confidence after three quarters with an annual Roadmap, then more radical alternatives SHOULD be pursued that satisfy the requirements from Oversight. We SHOULD also assess whether ZenHub is meeting our needs for a common project management application. 
+
+### Unresolved Questions
+
+*What aspects of the design do you expect to clarify further through the RFC review process?*
+
+The authors are seeking recommendations for:
+* Improving the accountability for Release schedule and deliverables.
```

Comment:
How would you feel about requiring a post-mortem to be conducted/filed if deliverables aren't met in time? This involves both a "post-mortem RFC" (what was expected to happen, what happened, why it happened, how do we prevent it in the future) and also a presentation of the post-mortem?

Reply:
Clarifying: for each objective/deliverable? Also, if deliverable means objective, can we try to minimize jargon here?

Reply:
_deliverables_ is simply short hand for the _acceptance criteria_ or _definition of done_ for an objective modeled as an epic or a spike. There may be multiple deliverables for one objective.

Reply:
So maybe one per objective?

Reply:
I could clarify the request as _Improving the accountability and confidence for on-time delivery of Roadmap Objectives during a Release. (In other words, what are the consequences for slipping the original Milestone or Release assigned to an Objective? This should not be friction-less.)_ 

In Industry, there are generally negative consequences for failing to hit your dates. Certainly, all my OKR(s) included the phrase "on time and at quality" where both "time" and "quality" were well documented.

Reply:
During milestone completion we should report on what we hit and what had to be consciously dropped. A "why" for each of these would be helpful during our report outwards and to oversight. Perhaps some of this could be built into the milestone completion process?

For post-mortems, i think of these as retros for acute, unplanned work and require more substantial documentation.

Reply:
@tj - does aa495d1dd798db2adbe970a996989b277d3da92f address your suggestion?

Reply:
This looks good to me thanks!